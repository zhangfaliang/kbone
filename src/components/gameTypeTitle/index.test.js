import GameTypeTitle from '.';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
const baseketball = require('../../assets/game/index/basketball.png');
const baseketballActive = require('../../assets/game/index/basketballActive.png');
const soccer = require('../../assets/game/index/Soccer.png');
const soccerActive = require('../../assets/game/index/SoccerActive.png');

describe('Button组件', async () => {
  it('basic use', () => {
    expect.assertions(1);
    const wrapper = render(<GameTypeTitle prefixCls={null} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('checkShouldComponentUpdate', () => {
    expect.assertions(1);
    const gameTypeTitle = mount(<GameTypeTitle />);
    gameTypeTitle.setProps({
      activeKey: 'Basketball',
    });
    expect(gameTypeTitle.state('activeKey')).toBe(gameTypeTitle.prop('activeKey'));
  });

  it('clickDefaultCheckBtnGameTypeTitle', () => {
    expect.assertions(1);
    const gameTypeTitle = mount(<GameTypeTitle />);
    const handleGameType = gameTypeTitle.prop('handleGameType');
    const click = jest.fn(handleGameType);
    expect(click()).toBe('click');
  });

  it('clickCustomCheckBtnGameTypeTitle', () => {
    expect.assertions(1);
    const clickCheckBtn = () => 'clickCheckBtn';
    const click = jest.fn(clickCheckBtn);
    const gameTypeTitle = mount(<GameTypeTitle handleGameType={click} />);
    gameTypeTitle
      .find('GameTypeTitle')
      .at(0)
      .find('.items')
      .at(0)
      .simulate('click');
    expect(click).toHaveReturnedWith('clickCheckBtn');
  });
});
