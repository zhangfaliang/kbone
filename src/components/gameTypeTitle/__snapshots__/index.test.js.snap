// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button组件 basic use 1`] = `
<div
  class="default-game-type-title"
>
  <div
    class="items"
  >
    <img
      alt="01"
      class="img"
      src="[object Object]"
    />
    <span
      class="typeText activeTypeText"
    >
      Soccer (22)
    </span>
  </div>
  <div
    class="items"
  >
    <img
      alt="02"
      class="img"
      src="[object Object]"
    />
    <span
      class="typeText"
    >
      Basketball (22)
    </span>
  </div>
</div>
`;
